### PR TITLE
[luci] Revise FuseActivationFunctionPass test

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise FuseActivationFunctionPass test with TestIOGraph.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6110, from #6111

File: compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp [hunk 1]
```diff
@@ -43,32 +45,97 @@ namespace
  *         [Conv2]
  *
  */
-class SimpleGraph
+class ConvReluConvGraphlet
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp:48]: Change name to `ConvReluConvGraphlet` to reflect nodes

File: compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp [hunk 2]
```diff
@@ -43,32 +45,97 @@ namespace
...
+  luci::CircleConv2D *_conv1 = nullptr;
+  luci::CircleConv2D *_conv2 = nullptr;
+  luci::CircleRelu *_relu = nullptr;
+  luci::CircleConst *_conv1_f = nullptr;
+  luci::CircleConst *_conv1_b = nullptr;
+  luci::CircleConst *_conv2_f = nullptr;
+  luci::CircleConst *_conv2_b = nullptr;
+};
+
+class FuseActTestGraph : public TestIOGraph, public ConvReluConvGraphlet
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp:89]: Added for actual testing

File: compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp [hunk 3]
```diff
@@ -43,32 +45,97 @@ namespace
...
+    _conv2->bias(_conv2_b);
 
-    conv1->name("conv1");
-    conv2->name("conv2");
-    relu->name("relu");
+    output()->from(_conv2);
   }
+};
 
+class ConvHasMultiSuccGraph : public TestIOGraph, public ConvReluConvGraphlet
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp:113]: Also added for actual testing

File: compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp [hunk 4]
```diff
@@ -80,34 +147,46 @@ TEST(FuseActivationFunctionPassTest, name)
...
+
+  g.init();
 
-  // Conv1 has multiple successors
-  EXPECT_FALSE(luci::fuse_activation_function(g.relu));
+  // Relu input Conv2D has multiple successors
+  EXPECT_FALSE(pass.run(g.g()));
+}
+
+TEST(FusePreActivationBatchNorm, fuse_activation_function_tanh_NEG)
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseActivationFunctionPass.test.cpp:182]: Test case separated for `TANH`

**Review by mhs4670go — DISMISSED** [review]
LGTM

**Comment by seanshpark** [comment]
CI build failure depends on #6206

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by llFreetimell — APPROVED** [review]
LGTM

**Review by jinevening — APPROVED** [review]

**Pull request merged**